### PR TITLE
Change  Replidog::Model#lock!

### PR DESCRIPTION
**Pull request by Y-Fujikawa**
I using ActiveRecord in with_lock.
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/locking/pessimistic.rb#L81-L86
But, Used replidog with with_lock to Occurred `ArgumentError`.
```ruby
ArgumentError: wrong number of arguments (given 1, expected 0)
```

I check replidog in source.
Nothing `with_lock` method.
So make `with_lock` method.

Please review this pull request.

### CODE REVIEW

**Comment by jemiam** [comment]
The error seemed to be caused because Replidog::Model#lock! doesn't support necessary arguments.

In replidog all methods not defined in Replidog::Model are delegated to ActiveRecord::Base via [method_missing](https://github.com/jemiam/replidog/blob/v1.1.2/lib/replidog/proxy.rb#L107-L113). So i should implement lock! method with the correct signature.

I'll merge this PR if you fix the problem in the proper way. 

Thanks for pointing out my mistakes anyway 😃

File: spec/replidog/model_spec.rb [hunk 1]
```diff
@@ -304,4 +304,25 @@ def activate_connection_pools
...
     end
   end
+
+  describe "#with_lock" do
+    context "with using master name" do
+      it "executes SQL query on master connection" do
+        Recipe.using(:master).create(title: "test")
+        Recipe.using(:master) do
+          Recipe.first.with_lock do
+            Recipe.first.update!(title: "test_update")
```
Comment (jemiam) [inline spec/replidog/model_spec.rb]: Please change `update!` to ` update_attributes!`.
Rails 3.x doesn't seem to implement `update!` method.

**Comment by jemiam** [comment]
You don't need to add with_lock method, just change the signature of lock! method like this. 

```
$ git diff lib/replidog/model.rb
diff --git a/lib/replidog/model.rb b/lib/replidog/model.rb
index 57b5488..b937631 100644
--- a/lib/replidog/model.rb
+++ b/lib/replidog/model.rb
@@ -54,7 +54,7 @@ def clear_all_connections!
     end

     module InstanceMethodsWithReplidogSupport
-      def lock!
+      def lock!(lock = true)
         old_connection_name = self.class.connection.current_connection_name
         self.class.connection.current_connection_name ||= :master
         super
```

英語下手でうまく伝わってなかったみたいですね 😅

**Comment by Y-Fujikawa** [comment]
拙い英語の文章となりすみません 🙇 
修正内容、理解しました  👍

**Pull request merged**